### PR TITLE
[e2e] custom environment example

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

Add example of a custom e2e test setup, including

* one Amazon Linux host
* lighttpd installed on docker on the host
* agent installed on host
* fakeintake deployed to ECS

### Motivation

There is a lack of documentation of how to setup a custom environment: this new example covers few remote host features, including file system operations, docker compose, agent install.

It also includes the `Initializable` implementation to initialise clients.

### CODE REVIEW

File: test/new-e2e/examples/customenv_with_docker_app_test.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+			"DD_LIGHTTPD_CONFIG":    pulumi.String(lighttpdConfigFile),
+			"DD_LIGHTTPD_LOGS_PATH": pulumi.String(lighttpdDir),
+		}
+		// compose lighttpd
+		composeLighttpdCmd, err := dockerManager.ComposeStrUp("lighttpd", []docker.ComposeInlineManifest{
+			{
+				Name:    "lighttpd",
+				Content: pulumi.String(lighttpdComposeContent),
+			},
+		}, envVars, pulumi.DependsOn(mountedFilesCommands))
```
Comment (pducolin) [inline test/new-e2e/examples/customenv_with_docker_app_test.go:144]: **💭 thought**
We could simplify this removing the name parameter and using a concat of compose manifest names

File: test/new-e2e/examples/customenv_with_docker_app_test.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+		// replace `ACCESS_LOG_PATH` and `ERROR_LOG_PATH` in the integration config
+		lighttpdIntegrationConfigContent = strings.ReplaceAll(lighttpdIntegrationConfigContent, "LIGHTTPD_LOG_PATH", lighttpdDir)
+		// install the agent on the remote host
+		agent, err := agent.NewHostAgent(awsEnv.CommonEnvironment, remoteHost,
+			agentparams.WithFakeintake(fakeIntake),
+			agentparams.WithIntegration("lighttpd.d", lighttpdIntegrationConfigContent),
+			agentparams.WithLogs(),
+			// agent depends on the docker compose command
+			agentparams.WithPulumiResourceOptions(pulumi.DependsOn([]pulumi.Resource{composeLighttpdCmd})),
+		)
```
Comment (pducolin) [inline test/new-e2e/examples/customenv_with_docker_app_test.go:157]: **💭 thought**
`agent` could allow `Export` to `components.Agent`

File: test/new-e2e/examples/customenv_with_docker_app_test.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+	// initialize agent host
+	agent, err := client.NewHostAgentClient(ctx.T(), e.RemoteHost, true)
+	if err != nil {
+		return err
+	}
+	e.Agent.Client = agent
+
+	return nil
+}
```
Comment (pducolin) [inline test/new-e2e/examples/customenv_with_docker_app_test.go:66]: **💭 thought**
we might get rid of this

**Review by FlorentClarret — APPROVED** [review]
Nice!

**Pull request merged**